### PR TITLE
Add context menu for block editors

### PR DESCRIPTION
**Pull request by Hamms**
Adds a right-click-accessible context menu enabled only in block edit
mode to set values like uservisible, movable, deletable, and others that
were previously only editable through raw XML.

untitled

https://www.pivotaltracker.com/n/projects/1256652/stories/117876041

### CODE REVIEW

**Comment by Hamms** [comment]
Note that this is less "adding a context menu" and more "reenabling and modifying a long-disabled context menu"

**Comment by bcjordan** [comment]
LGTM!

**Comment by Hamms** [comment]
@bcjordan could you take a quick look at one cleanup change I forgot to push?

File: blockly-core/core/ui/contextmenu.js [hunk 1]
```diff
@@ -136,16 +133,6 @@ Blockly.ContextMenu.optionToDom_ = function(text) {
 };
 
 /**
- * Hide the context menu.
- */
-Blockly.ContextMenu.hide = function() {
```
Comment (bcjordan) [inline blockly-core/core/ui/contextmenu.js:141]: @Hamms is this somewhere else now / was it duplicated?

Comment (bcjordan) [inline blockly-core/core/ui/contextmenu.js:141]: Ah yup, looks like there is another one in there already. 💁

👍

**Comment by bcjordan** [comment]
LGTM

File: blockly-core/core/ui/contextmenu.js (hunk 1 above)
Comment (Hamms) [inline blockly-core/core/ui/contextmenu.js:141]: hah, yeah, I'm not sure why this was added but it definitely shouldn't have been.

Unfortunately, none of this stuff has been touched this the current repo was created.

**Pull request merged**